src/main.rs

```diff
@@ -191,7 +191,8 @@ fn run_post_build_script() -> Option<process::ExitStatus> {
     let build_script_manifest_path = build_script_manifest_dir.join("Cargo.toml");
     let build_script_manifest_content = format!(
         include_str!("post_build_script_manifest.toml"),
-        file_name = post_build_script_path.display(),
+        file_name = toml::to_string(&post_build_script_path.to_str())
+            .expect("Failed to write post build script manifest"),
```

Comment:
This error message is a bit misleading since we don't try to write any manifest file yet at this point. How about:

```suggestion
            .expect("Failed to serialize post build script path as TOML string"),
```